packages/fast-form-generator-react/src/form/form-category.tsx

```diff
@@ -0,0 +1,136 @@
+
+/**
+ * Select state interface
+ */
+export interface IFormCategoryProps {
+
+    /**
+     * Passes the category item
+     */
+    categoryItem: JSX.Element[];
+
+    /**
+     * Passes the category title
+     */
+    title: string;
+
+    /**
+     * Is the category expandable
+     */
+    Expandable?: boolean;
+
+    /**
+     * Is the category expanded
+     */
+    isExpanded?: boolean;
+
+    /**
+     * Passes the id
+     */
+    id?: string;
+}
+
+export interface IFormCategoryState {
+    isCategoryExpanded: boolean;
+}
+
+/**
+ * Schema form component definition
+ * @extends React.Component
+ */
+/* tslint:disable-next-line */
+class FormCategory extends React.Component<IFormCategoryProps & IManagedClasses<IFormCategoryClassNameContract>, IFormCategoryState> {
+
+    constructor(props: IFormCategoryProps & IManagedClasses<IFormCategoryClassNameContract>) {
+        super(props);
+
+        if (this.props.Expandable) {
+            this.state = {
+                isCategoryExpanded: this.props.isExpanded,
+            };
+        }
+    }
+    /**
+     * Renders the component
+     */
+    public render(): JSX.Element {
+        // Exit if the array is only one item long and that item is null or undefined
+        if (this.props.categoryItem.length < 1 && this.props.categoryItem[0] === null ||
+            this.props.categoryItem.length < 1 && this.props.categoryItem[0] === undefined) {
+            return;
+        }
+
+        return (
+            <div key={this.props.title}>
+                {this.renderHeader()}
+                <div
+                    className={this.getClassNames()}
+                    {...this.generateContainerAttributes()}
+                >
+                    {this.props.categoryItem}
+                </div>
+            </div>
+        );
+    }
+
+    private generateContainerAttributes(): React.HtmlHTMLAttributes<HTMLDivElement> {
+        if (!this.props.Expandable) {
+            return;
+        }
+
+        const attributes: Partial<React.HtmlHTMLAttributes<HTMLDivElement>> = {};
+        attributes["aria-hidden"] = !this.state.isCategoryExpanded;
+        attributes.id = this.props.id;
+
+        return attributes;
+    }
+
+    private handleCategoryCollapse = (): void => {
+        this.setState({
+            isCategoryExpanded: !this.state.isCategoryExpanded
+        });
+    }
+
+    private getClassNames(): string {
+        const classNames: string = this.props.Expandable
+            ? this.state.isCategoryExpanded
+            ? this.props.managedClasses.formCategory
+            : this.props.managedClasses.formCategory__collapsed
+            : this.props.managedClasses.formCategory;
+
+        return classNames;
+    }
+
+    private renderHeader(): JSX.Element {
+        const header: JSX.Element = this.props.Expandable
+        ? {...this.renderHeaderButton()}
+        : {...this.renderHeaderTitle()};

```

Comment:
Instead of creating a single function here, can we just reference these inline in place?

```
{ this.props.expandable ? this.renderHeaderButton() : this.renderHeaderTitle() }
```

Reply:
+1 for @chrisdholt suggestion.